src/build_tools.rs

```diff
@@ -148,15 +166,37 @@ impl SchemaError {
 impl SchemaError {
     #[new]
     fn py_new(message: String) -> Self {
-        Self { message }
+        Self {
+            value: SchemaErrorInner::Message(message),
+        }
+    }
+
+    fn __repr__(&self, py: Python) -> String {
+        match &self.value {
+            SchemaErrorInner::Message(message) => format!("SchemaError({message:?})"),
+            SchemaErrorInner::Error(error) => error.display(py),
+        }
     }
 
-    fn __repr__(&self) -> String {
-        format!("{self:?}")
+    fn __str__(&self, py: Python) -> String {
+        match &self.value {
+            SchemaErrorInner::Message(message) => message.to_owned(),
+            SchemaErrorInner::Error(error) => error.display(py),
+        }
     }
 
-    fn __str__(&self) -> String {
-        self.to_string()
+    fn error_count(&self) -> usize {
+        match &self.value {
+            SchemaErrorInner::Message(_) => 0,
+            SchemaErrorInner::Error(error) => error.error_count(),
+        }
+    }
+
+    fn errors(&self, py: Python) -> PyResult<Py<PyList>> {
+        match &self.value {
+            SchemaErrorInner::Message(_) => Ok(PyList::new(py, Vec::<PyAny>::new()).into_py(py)),
```

Comment:
not sure, but I imagine it could be confusing if there's an error, but the number of errors is 0, and `.errors() == []`

Might be better to construct a dict here which matches the typeddict and includes the message?